### DISCUSSION TITLE
Error 0x80091007: Failed to cache payload

### DISCUSSION THREAD

**Discussion by mounikapk** [wixtoolset/issues#7299] [Questions]
Hi WiX users,

I have been seeing the below issue with our MSI when we are doing a new install of our product .

***************************************************************************************************
 Acquiring package:  OURPRODUCT  , payload: OURPRODUCT, copy from: C:\23_test\products\ourprod\SetUps\OURPRODUCT\OurProductSetup.msi
[0F3C:0FB4][2023-03-21T16:04:33]: Error 0x80091007: Hash mismatch for path: C:\ProgramData\Package Cache\.unverified\ OURPRODUCT
[0F3C:0FB4][2023-03-21T16:04:33]: Error 0x80091007: Failed to verify hash of payload:  OURPRODUCT
[0F3C:0FB4][2023-03-21T16:04:33]: Failed to verify payload:  OURPRODUCT  at path: C:\ProgramData\Package Cache\.unverified\ OURPRODUCT , error: 0x80091007. Deleting file.
[0F3C:0FB4][2023-03-21T16:04:33]: Error 0x80091007: Failed to cache payload: OURPRODUCT
***************************************************************************************************

We use a Custom Bootstrapper built on **WiX Toolset 3.6** with MSI Chaining. We have multiple MsiPackages with in the bundle.

Additional details-
1. We haven't changed anything in the bootstrapper since it was working with our previous version of the product. Only added one new payload to the bundle as below-
<Payload Id="B25AAB41CF014F838958ECBE1B10A747" SourceFile ="$(var.SolutionDir)\..\3rdPartyTools\Oracle.ManagedDataAccess.dll"></Payload>

2. We are neither signing the bundle nor the MSIs. 
3. We are not changing / modifying the bundle after the bundle is generated
4. We collect the bundle and build an .iso out of it and use this .iso which has the bundle (setup.exe) to install the application.

When we execute the MSI separately, it installs just fine.
We have also set the Product Id="*" for all our MSIs (as per https://github.com/orgs/wixtoolset/discussions/6569 )

Any clues as to what needs to be checked or what might cause this error? Any help shall be greatly appreciated.

Regards,
Mounika.

**Comment by robmen** [comment]
@mounikapk I meant hash the MSI file that was downloaded by Burn to see if it matched the hash that Burn expected. First thing to rule out is if the payload got corrupted in transit (which is what the error message would suggest).

**Reply by mounikapk** [reply]
@robmen , I am double checking if the payload is corrupt b y any chance. Thanks again!
